csharp/FileManager/Manager.cs

```diff
@@ -0,0 +1,138 @@
+
+        public Manager(string DBFilename)
+        {
+            var linksConstants = new LinksConstants<TLinkAddress>(enableExternalReferencesSupport: true);
+            var dataMemory = new FileMappedResizableDirectMemory(DBFilename);
+            Links = new UnitedMemoryLinks<UInt64>(dataMemory, UnitedMemoryLinks<UInt64>.DefaultLinksSizeStep , linksConstants, IndexTreeType.Default);
+            var link = Links.Create();
+            link = Links.Update(link, newSource: link, newTarget: link);
+            ushort currentMappingLinkIndex = 1;
+            _meaningRoot = GetOrCreateMeaningRoot(currentMappingLinkIndex++);
+            _unicodeSymbolMarker = GetOrCreateNextMapping(currentMappingLinkIndex++);
+            _unicodeSequenceMarker = GetOrCreateNextMapping(currentMappingLinkIndex++);
+            _meaningRoot = GetOrCreateMeaningRoot(currentMappingLinkIndex++);
+            _addressToNumberConverter = new AddressToRawNumberConverter<TLinkAddress>();
+            _numberToAddressConverter = new RawNumberToAddressConverter<TLinkAddress>();
+            var balancedVariantConverter = new BalancedVariantConverter<TLinkAddress>(Links);
+            var unicodeSymbolCriterionMatcher = new TargetMatcher<TLinkAddress>(Links, _unicodeSymbolMarker);
+            var unicodeSequenceCriterionMatcher = new TargetMatcher<TLinkAddress>(Links, _unicodeSequenceMarker);
+            var charToUnicodeSymbolConverter = new CharToUnicodeSymbolConverter<TLinkAddress>(Links, _addressToNumberConverter, _unicodeSymbolMarker);
+            var unicodeSymbolToCharConverter = new UnicodeSymbolToCharConverter<TLinkAddress>(Links, _numberToAddressConverter, unicodeSymbolCriterionMatcher);
+            var sequenceWalker = new RightSequenceWalker<TLinkAddress>(Links, new DefaultStack<TLinkAddress>(), unicodeSymbolCriterionMatcher.IsMatched);
+            _stringToUnicodeSequenceConverter = new CachingConverterDecorator<string, TLinkAddress>(new StringToUnicodeSequenceConverter<TLinkAddress>(Links, charToUnicodeSymbolConverter, balancedVariantConverter, _unicodeSequenceMarker));
+            _unicodeSequenceToStringConverter = new CachingConverterDecorator<TLinkAddress, string>(new UnicodeSequenceToStringConverter<TLinkAddress>(Links, unicodeSequenceCriterionMatcher, sequenceWalker, unicodeSymbolToCharConverter));
+        }
+
+        private TLinkAddress GetFileLink(string name)
+        {
+            var nameLink = _stringToUnicodeSequenceConverter.Convert(name);
+            var any = Links.Constants.Any;
+            var query = new Link<UInt64>(index: any, source: nameLink, target: any);
+            var list = new List<IList<TLinkAddress>>();
+            var listFiller = new ListFiller<IList<TLinkAddress>, TLinkAddress>(list, Links.Constants.Continue);
+            Links.Each(listFiller.AddAndReturnConstant, query);
+            if (list.Count != 0)
+            {
+                return list.First()[Links.Constants.TargetPart];

```

Comment:
This can be replaced with `Links.GetTarget(list.First())` for better readability (aka shorter code).